doc/api.rst

```diff
@@ -0,0 +1,478 @@
+    given, it is converted to a :py:class:`Mean` object using 
+    :py:meth:`Mean.__init__`.
+
+    .. py:attribute:: covar
+        :type: InlineSection
+
+    Stores the covariance parameters for named distributions.
+    
+    Behaves as a dictionary where keys are distribution names and each
+    value should be set to the covariance value or a :py:class:`Covar` object 
+    initialized with the covariance value. If a Python float or list of floats is 
+    given, it is converted to a :py:class:`Covar` object using 
+    :py:meth:`Covar.__init__`.
+
+    .. py:attribute:: dpmf
+        :type: InlineSection
+
+    Stores dense probability mass dunction (DPMF) objects, which can later be used
+    to define Gaussian Mixture models. 
+    
+    Behaves as a dictionary where keys are distribution names and each value 
+    should be set to a :py:class:`DPMF` object. 
+
+    .. py:attribute:: mc
+        :type: InlineMCSection
+        :value: InlineMCSection(mean = self.mean, covar = self.covar)
+
+    Stores Gaussians acting as mixture components (MC) for a Gaussian mixture
+    model.
+    
+    Behaves as a dictionary where keys are distribution names and each value 
+    should be set to an :py:class:`MC` object.
+
+    .. py:attribute:: mx
+        :type: InlineMXSection
+    
+    Store Gaussian mixture (mx) distributions constructed from above-defined mixture 
+    components and dense probability mass functions.
+    
+    Behaves as a dictionary where keys are distribution names, usually 
+    corresponding to hidden state names of an emission variable (from 
+    :py:attr:`self.name_collection`) and each value is an :py:class:`MX` object.
+
+    .. py:attribute:: dense_cpt
+        :type: InlineSection
+
+    Stores dense conditional probability tables (CPTs) used in the model. 
+    
+    Behaves as a dictionary where keys are distribution names, which can 
+    be referenced in the structure file, and each value is a 
+    :py:class:`DenseCPT` object.  
+
+    .. py:attribute:: deterministic_cpt
+        :type: InlineSection
+
+    Stores deterministic conditional probability tables (CPTs) used in the model.
+    
+    Behaves as a dictionary where keys are distribution names, which can 
+    be referenced in the structure file, and each value is a 
+    :py:class:`DeterministicCPT` object.
+
+    .. py:method:: __init__(self)
+
+        Create an `InputMaster` object where all attributes are empty.
+
+    .. py:method:: save(self, filename)
+
+        Save all parameters to the provided file, for Segway to use in training
+        and annotation.
+
+        :param filename: Path to input master file, where results are saved
+        :type filename: str
+        :returns: None
+        :rtype: None
+    
+Usage example:
+
+.. code-block:: python
+
+    # Create InputMaster object
+    input_master = InputMaster()
+    # Set parameters
+    ...
+    # Save to output file
+    input_master.save(outfile)
```

Comment:
probably rename `outfile` to something more filename-ish instead of a potential file handle.